lib/matplotlib/patches.py

```diff
@@ -71,7 +72,9 @@ def __init__(self, *,
         if joinstyle is None:
             joinstyle = JoinStyle.miter
 
-        self._hatch_color = colors.to_rgba(mpl.rcParams['hatch.color'])
+        self._original_edgecolor = None
+        self._original_hatchcolor = None
```

Comment:
As mentioned in another comment, even for patches I initialize the `_original_edgecolor` and `_original_hatchcolor` to be None, so that I can check whether edgecolor/hatchcolor has been set. This is here, because it has to be initialized, before calling the `set_edgecolor()` and `set_hatchcolor()` function.

Reply:
Do we really need this "original" logic. Couldn't we instead leave the hatchcolor value as "inherit" and only resolve the actual color before drawing?